table/tables/partition.go

```diff
@@ -1108,6 +1151,33 @@ func (lp *ForListColumnPruning) LocateRanges(sc *stmtctx.StatementContext, r *ra
 		locations = append(locations, item.location)
 		return true
 	})
+	if lp.HasDefault() {
+		// Add the default partition since there may be a gap
+		// between the conditions range and the LIST COLUMNS values
+		addDefault := true
+		if len(lp.tblInfo.Partition.Columns) == 1 {
+			rangeSteps := 0
+			for ; rangeSteps <= len(locations); rangeSteps++ {
+				if bytes.Compare(lowKey, highKey) < 0 {
+					// TODO: Does this work? Also for int?
+					lowKey = kv.Key(lowKey).PrefixNext()
+				} else {
+					break
+				}
+			}
+			if rangeSteps == len(locations) {
+				addDefault = false
+			}
+		}
```

Comment:
`addDefault` will always be true, so we can remove this check and always append default partition here.